src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -1148,7 +1228,7 @@ internal Type TypeForInitialization
             //[SecuritySafeCritical]
             private void SetTypeForInitialization(Type classType)
             {
-                //if (classType.IsDefined(Globals.TypeOfDataContractAttribute, false))
+                //if (classType.IsSerializable || classType.IsDefined(Globals.TypeOfDataContractAttribute, false))
```

Comment:
I tried uncommenting this. But I got NullReferenceException at other place. I assumed we need to initialize _typeForInitialization in NetCore for some reason, so I kept this as is.